metrics/event.trx/trx.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2022 Block, Inc.
+
+package eventtrx
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"strconv"
+
+	"github.com/cashapp/blip"
+	"github.com/cashapp/blip/sqlutil"
+)
+
+const (
+	DOMAIN           = "event.trx"
+	OLDEST_TRX_QUERY = `SELECT UNIX_TIMESTAMP(NOW()) - UNIX_TIMESTAMP(MIN(trx_started)) AS time FROM information_schema.innodb_trx;`
+)
+
+type trxMetrics struct {
+	queryOldest bool
+}
+
+// Trx collects metrics for the event.trx domain.
+// The source is information_schema.innodb_trx.
+type Trx struct {
+	db      *sql.DB
+	atLevel map[string]trxMetrics
+}
+
+// Verify collector implements blip.Collector interface
+var _ blip.Collector = &Trx{}
+
+// NewTrx makes a new Trx collector.
+func NewTrx(db *sql.DB) *Trx {
+	return &Trx{
+		db:      db,
+		atLevel: map[string]trxMetrics{},
+	}
+}
+
+// Domain returns the Blip metric domain name (DOMAIN const).
+func (c *Trx) Domain() string {
+	return DOMAIN
+}
+
+// Help returns the output for blip --print-domains.
+func (c *Trx) Help() blip.CollectorHelp {
+	return blip.CollectorHelp{
+		Domain:      DOMAIN,
+		Description: "Event transaction metrics from information_schema.innodb_trx",
```

Comment:
Let's go with just "Transaction metrics"  because trx are a type of even (they're not "event transactions" per-se) and Blip tries to hide implementation details like information_schema.innodb_trx so they can vary (as we discussed, it'd technically be nicer to use perf_schema for this).

Reply:
Done